includes/elementor-functions.php

```diff
@@ -0,0 +1,222 @@
+
+	$header_width = hello_elementor_get_setting( 'hello_header_width' );
+	if ( 'full-width' === $header_width ) {
+		$layout_classes[] = 'header-full-width';
+	}
+
+	$header_menu_dropdown = hello_elementor_get_setting( 'hello_header_menu_dropdown' );
+	if ( 'tablet' === $header_menu_dropdown ) {
+		$layout_classes[] = 'menu-dropdown-tablet';
+	} elseif ( 'mobile' === $header_menu_dropdown ) {
+		$layout_classes[] = 'menu-dropdown-mobile';
+	} elseif ( 'none' === $header_menu_dropdown ) {
+		$layout_classes[] = 'menu-dropdown-none';
+	}
+
+	$hello_header_menu_layout = hello_elementor_get_setting( 'hello_header_menu_layout' );
+	if ( 'dropdown' === $hello_header_menu_layout ) {
+		$layout_classes[] = 'menu-layout-dropdown';
+	}
+
+	return implode( ' ', $layout_classes );
+}
+
+/**
+ * Helper function to translate the footer layout setting into a class name.
+ *
+ * @return string
+ */
+function hello_get_footer_layout_class() {
+	$footer_layout = hello_elementor_get_setting( 'hello_footer_layout' );
+
+	$layout_classes = [];
+
+	if ( 'inverted' === $footer_layout ) {
+		$layout_classes[] = 'footer-inverted';
+	} elseif ( 'stacked' === $footer_layout ) {
+		$layout_classes[] = 'footer-stacked';
+	}
+
+	$footer_width = hello_elementor_get_setting( 'hello_footer_width' );
+
+	if ( 'full-width' === $footer_width ) {
+		$layout_classes[] = 'footer-full-width';
+	}
+
+	if ( hello_elementor_get_setting( 'hello_footer_copyright_display' ) && '' !== hello_elementor_get_setting( 'hello_footer_copyright_text' ) ) {
+		$layout_classes[] = 'footer-has-copyright';
+	}
+
+	return implode( ' ', $layout_classes );
+}
+
+add_action( 'elementor/editor/after_enqueue_scripts', function() {
+	$suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
+
+	wp_enqueue_script(
+		'hello-theme-editor',
+		get_template_directory_uri() . '/assets/js/hello-editor' . $suffix . '.js',
+		[ 'jquery', 'elementor-editor' ],
+		HELLO_ELEMENTOR_VERSION,
+		true
+	);
+
+	wp_enqueue_style(
+		'hello-editor',
+		get_template_directory_uri() . '/editor' . $suffix . '.css',
+		[],
+		HELLO_ELEMENTOR_VERSION
+	);
+} );
+
+add_action( 'wp_enqueue_scripts', function() {
+	$suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
+
+	wp_enqueue_script(
+		'hello-theme-editor',
+		get_template_directory_uri() . '/assets/js/hello-frontend' . $suffix . '.js',
+		[ 'jquery' ],
+		'1.0.0',
+		true
+	);
+
+	if ( did_action( 'elementor/loaded' ) ) {
+		// Load the kit site-wide
+		Elementor\Plugin::$instance->kits_manager->frontend_before_enqueue_styles();
+	}
+
+} );
+
+
+/**
+ * Helper function to decide whether to output the header template.
+ *
+ * @return bool
+ */
+function hello_get_header_display() {
+	$is_editor = isset( $_GET['elementor-preview'] );
+
+	return (
+		hello_elementor_get_setting( 'hello_header_logo_display' )
+		|| hello_elementor_get_setting( 'hello_header_tagline_display' )
+		|| hello_elementor_get_setting( 'hello_header_menu_display' )
+		|| $is_editor
```

Comment:
swap

Reply:
Move `$is_editor` to the be the first condition

Reply:
In all places

Reply:
We've moved it to the first condition, in which case we return true immediately avoiding a database call on the other options